### PR TITLE
feature(client): Add edit icon and bright overlay on the banner

### PR DESCRIPTION
**Pull request by lokeswaran-aj**
Close: #4457 

## PR Details

- Added Edit icon when hovered on the Resource name
- Added a bright overlay on the banner on hover

https://user-images.githubusercontent.com/74011196/227755309-abcbe4e2-bf8b-43fc-9b5e-f1a0d4b2adf2.mov

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

**Comment by lokeswaran-aj** [comment]
Can someone approve this PR to run the checks.

@souravjain540

**Comment by yuval-hazaz** [comment]
@GreenMachine01 please review the video and see if that what you expected with the overlay
my opinion is that the overlay should only be seen in "edit" mode and not on hover, but it is up to you and @alexbass86

**Comment by lokeswaran-aj** [comment]
@GreenMachine01 what do you think about the overlay?

**Comment by GreenMachine01** [comment]
@lokeswaran-aj looks good.
let's proceed.

File: packages/amplication-client/src/Resource/ResourceHome.scss [hunk 1]
```diff
@@ -28,6 +28,19 @@ $circle-badge-border: 2px;
...
   }
+  &__header:hover::before {
+    z-index: 1;
+    content: "";
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    background-color: #ffffff99;
```
Comment (morhag90) [inline packages/amplication-client/src/Resource/ResourceHome.scss:39]: @lokeswaran-aj  Please change the hard-coded color value to var,
Example: var(--black20);

Comment (lokeswaran-aj) [reply packages/amplication-client/src/Resource/ResourceHome.scss]: I could not find an existing variable, so I have create a variable in the top and used it.

**Comment by souravjain540** [comment]
@morhag90 is it okay to go ahead and approve this PR?

**Review by mulygottlieb — APPROVED** [review]

**Comment by mulygottlieb** [comment]
@lokeswaran-aj I merged it, it will appear in the next release. Thank you for your contribution!

**Pull request merged**